scaleway/resource_iot_hub_device.go

```diff
@@ -0,0 +1,340 @@
+				Required:    true,
+				Description: "The name of the device",
+			},
+			"allow_insecure": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Allow plain and server-authenticated SSL connections in addition to mutually-authenticated ones",
+			},
+			"message_filters": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Description: "Filter-sets to authorize or deny the device to publish/subscribe to specific topics",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"publish": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Filter-set to restrict topics the device can publish to",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"policy": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Publish message filter policy",
+										Default:     iot.DeviceMessageFiltersPolicyReject.String(),
+										ValidateFunc: validation.StringInSlice([]string{
+											iot.DeviceMessageFiltersPolicyAccept.String(),
+											iot.DeviceMessageFiltersPolicyReject.String(),
+										}, false),
+									},
+									"topics": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "List of topics in the set",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+						"subscribe": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Filter-set to restrict topics the device can subscribe to",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"policy": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Subscribe message filter policy",
+										Default:     iot.DeviceMessageFiltersPolicyReject.String(),
+										ValidateFunc: validation.StringInSlice([]string{
+											iot.DeviceMessageFiltersPolicyAccept.String(),
+											iot.DeviceMessageFiltersPolicyReject.String(),
+										}, false),
+									},
+									"topics": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "List of topics in the set",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"region": regionSchema(),
+			// Computed elements
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the device",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the device",
+			},
+			"crt": {
+				Type:        schema.TypeString,
+				Computed:    true,
```

Comment:
We should use https://www.terraform.io/docs/extend/best-practices/sensitive-state.html for sensitive data

Reply:
Done